### ISSUE TITLE
Typescript combineArray type definition does not support arbitrary lengths.

### ISSUE THREAD

**Issue by mrapogee** [cujojs/most#333]
The typescript definition for combineArray lacks a arbitrary length overload.

Adding the following underneath the other overloads fixes the issue.
```typescript
export function combineArray<V, R> (
  fn: (...items: V[]) => R,
  items: Stream<V>[]
): Stream<R>;
```

Also, in the Stream interface
```typescript
combineArray<V, R>(
  fn: (a: A, ...rest: V[]) => R,
  streams: Stream<V>[]
): Stream<R>;
```

**Comment by mrapogee** [comment]
Submitted #334

**Comment by briancavalier** [comment]
Closed by #334.  Thanks again!
